tools/sva/src/parser.js

```diff
@@ -0,0 +1,274 @@
+
+        token = this.next();
+        if (token.type !== TokenType.kEqual) {
+          this.error_ = token.line + ": expected = after result id";
+          return undefined;
+        }
+
+        token = this.next();
+      }
+
+      if (token.type !== TokenType.kOp) {
+        this.error_ = token.line + ": expected Op got " + token.type;
+        return undefined;
+      }
+
+      let name = token.data.name;
+      let data = this.getInstructionData(name);
+      let operands = [];
+      let result_type = undefined;
+
+      for (let operand of data.operands) {
+        if (operand.kind === "IdResult") {
+          if (result_id === undefined) {
+            this.error_ = token.line + ": expected result id";
+            return undefined;
+          }
+          let o = new AST.Operand(ast, result_id.data.name, "result_id",
+            result_id.data.val, []);
+          if (o === undefined) {
+            return undefined;
+          }
+          operands.push(o);
+        } else {
+          if (operand.quantifier === "?") {
+            if (this.nextIsNewInstr()) {
+              break;
+            }
+          } else if (operand.quantifier === "*") {
+            while (!this.nextIsNewInstr()) {
+              let o = this.extractOperand(ast, result_type, operand);
+              if (o === undefined) {
+                return undefined;
+              }
+              operands.push(o);
+            }
+            break;
+          }
+
+          let o = this.extractOperand(ast, result_type, operand);
+          if (o === undefined) {
+            return undefined;
+          }
+
+          // Store the result type away so we can use it for context dependent
+          // numbers if needed.
+          if (operand.kind === "IdResultType") {
+            result_type = ast.getType(o.name());
+          }
+
+          operands.push(o);
+        }
+      }
+
+      // Verify only GLSL extended instructions are used
+      if (name === "OpExtInstImport" && operands[1].value() !== "GLSL.std.450") {
+        this.error_ = token.line + ": Only GLSL.std.450 external instructions supported";
+        return undefined;
+      }
+
+      let inst = new AST.Instruction(name, data.opcode, operands);
+
+      ast.addInstruction(inst);
+    }
+    return ast;
+  }
+
+  getInstructionData(name) {
+    return this.grammar_["instructions"][name];
+  }
+
+  nextIsNewInstr() {
+    let n0 = this.peek(0);
+    if (n0.type === TokenType.kOp || n0.type === TokenType.kEOF) {
+      return true;
+    }
+
+    let n1 = this.peek(1);
+    if (n1.type === TokenType.kEOF) {
+      return false;
+    }
+    if (n0.type === TokenType.kResultId && n1.type === TokenType.kEqual)
+      return true;
+
+    return false;
+  }
+
+  extractOperand(ast, result_type, data) {
+    let t = this.next();
+
+    let name = undefined;
+    let kind = undefined;
+    let value = undefined;
+    let params = [];
+    if (data.kind === "IdResult" || data.kind === "IdRef"
```

Comment:
I think you'll need more cases, as in https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/text.cpp#L210 

There is also memory-semantics-ID, scope-id,  optional-id.

Reply:
Will flush this out post land as I create test cases.

Reply:
done.